Content.Omu.Server/HereticalTome/FascinationSystem.cs

```diff
@@ -1,44 +1,31 @@
-using Content.Omu.Server.HereticalTome;
 using Content.Omu.Server.HereticalTome.Components;
 using Content.Shared.Examine;
 
-namespace Content.Omu.Server.HereticalTome.Fascination;
-
-[ByRefEvent]
-public readonly record struct FascinationUpdateEvent();
+namespace Content.Omu.Server.HereticalTome;
 
 public sealed class FascinationSystem: EntitySystem
 {
-[Dependency] private readonly ILocalizationManager _loc = default!;
-
 public override void Initialize()
 {
     base.Initialize();
-    SubscribeLocalEvent<FascinationComponent, FascinationUpdateEvent>(UpdateDesc);
+
     SubscribeLocalEvent<FascinationComponent, ExaminedEvent>(OnExamined);
 }
 
-private void UpdateDesc(EntityUid uid, FascinationComponent comp, ref FascinationUpdateEvent Eventargs)
-{
-    switch(FascinationInt)
+    private void OnExamined(Entity<FascinationComponent> ent, ref ExaminedEvent args)
+    {
+        var comp = ent.Comp;
+        comp.FascinationDesc = comp.FascinationInt switch
         {
-            case 0: FascinationDesc = Loc.GetString("Fascination-0"); break;
-
-            case 1: FascinationDesc = Loc.GetString("Fascination-1"); break;
-
-            case 2: FascinationDesc = Loc.GetString("Fascination-2"); break;
-
-            case 3: FascinationDesc = Loc.GetString("Fascination-3"); break;
-
-            case 4: FascinationDesc = Loc.GetString("Fascination-4"); break;
-
-            case 5: FascinationDesc = Loc.GetString("Fascination-5"); break;
-        }
-
-}
-
-private void OnExamined(Entity<FascinationComponent> ent, ref ExaminedEvent args)
-{
-args.PushMarkup(FascinationDesc);
-}
+            0 => Loc.GetString("Fascination-0"),
+            1 => Loc.GetString("Fascination-1"),
+            2 => Loc.GetString("Fascination-2"),
+            3 => Loc.GetString("Fascination-3"),
+            4 => Loc.GetString("Fascination-4"),
+            5 => Loc.GetString("Fascination-5"),
+            _ => comp.FascinationDesc
+        };
+        args.PushMarkup(comp.FascinationDesc ?? Loc.GetString("Fascination-0"));
+        Dirty(ent);
```

Comment:
Dirty, stinky component go take a shower so we can update you manually onto the comp on the client